src/newrelic.ts

```diff
@@ -40,53 +65,361 @@ function addSegment(spec: Spec) {
 
       ctx.actdef.func.$$newrelic_wrapped$$ = true;
     }
-
-    Object.defineProperty(
-      ctx.actdef.func, 'name', { value: 'newrelic_' + origfunc.name })
   }
 }
 
+interface TelemetrySpecMetadata {
```

Comment:
move the type defs into a types.ts file, since we now have lots of them - see for example: https://github.com/jsonicjs/jsonic/blob/nextgen/types.ts